hypex/dataset/datagenerator.py

```diff
@@ -0,0 +1,293 @@
+        """
+        m_vector = m_with_dummy.dot(np.ones(self.num_features))
+        epsilon = np.random.multivariate_normal(mean=np.zeros(2),
+                                                cov=np.identity(2),
+                                                size=self.num_records)
+        y0 = m_vector + epsilon[:, 0]
+        y1 = self.sigma + m_with_dummy.dot(np.ones(self.num_features) * 2) + epsilon[:, 1]
+        return y0, y1
+
+    def _generate_outcome_and_full_data(self, m_with_dummy, treatment):
+        """
+        Generates outcome depending on treatment and create array with main numeric features
+        Args:
+            m_with_dummy: Array with main features
+            treatment: Array with treatment values
+
+        Returns:
+            data: Array with main numeric features
+        """
+        y0, y1 = self._generate_y_for_target(m_with_dummy)
+        if self.is_treatment and self.num_targets == 1:
+            outcome = (1 - treatment.flatten()) * y0 + treatment.flatten() * y1
+        else:
+            outcome = np.multiply((1 - treatment), y0[:, np.newaxis]) + np.multiply(treatment, y1[:, np.newaxis])
+            treatment = treatment[:, 0:1]
+
+        outcome = np.digitize(outcome, bins=np.array([np.mean(outcome) - 2])) if self.is_binary_target else outcome
+
+        if self.is_treatment:
+            return np.column_stack((m_with_dummy, treatment, outcome))
+        return np.column_stack((m_with_dummy, outcome))
+
+    def _set_df(self, data: np.ndarray, info: np.ndarray):
+        """
+        Generate names for columns and combine all data to DataFrame
+        Args:
+            data:
+                Array with main features
+            info:
+                Array with info features
+
+        Returns:
+            df:
+                DataFrame with generated data
+        """
+        gender = [["male", "female"][i] for i in np.random.choice(a=np.array([0, 1]), size=self.num_records)]
+        product = [['Deposit', 'Credit', 'Investment'][i] for i in np.random.choice(a=np.array([0, 1, 2]),
+                                                                                    size=self.num_records)]
+        self.treatment_name = ['treatment'] if self.is_treatment else []
+        df = pd.DataFrame(data=data, columns=self.features_names + self.treatment_name + self.target_names)
+        self.treatment_name = 'treatment' if self.is_treatment else []
+        for i in range(self.num_info_cols):
+            df.insert(i, self.info_col_names[i], pd.Series(info[i]))
+        df.insert(self.num_info_cols, 'feature_1', pd.Series(gender))
+        df.insert(self.num_info_cols + 1, 'feature_2', pd.Series(product))
+        self.features_names = ['feature_1', 'feature_2'] + self.features_names
+        return df
+
+    def _set_column_names(self):
+        """
+        Set names for columns in DataFrame
+        """
+        self.features_names = ['feature_' + str(i) for i in range(3, self.num_features + 3)]
```

Comment:
let's use f'string here. 
`self.features_names = [f'feature_{i}' for i in range(3, self.num_features + 3)]`